htdocs/core/triggers/interface_90_modSociete_ContactRoles.class.php

```diff
@@ -84,11 +84,7 @@ public function runTrigger($action, $object, User $user, Translate $langs, Conf
 				if (is_array($TContact) && ! empty($TContact)) {
 					$TContactAlreadyLinked = array();
 					if ($object->id > 0) {
-						$cloneFrom = dol_clone($object, 1);
-
-						if (! empty($cloneFrom->id)) {
-							$TContactAlreadyLinked = array_merge($cloneFrom->liste_contact(- 1, 'external'), $cloneFrom->liste_contact(- 1, 'internal'));
-						}
+						$TContactAlreadyLinked = array_merge($object->liste_contact(- 1, 'external'), $object->liste_contact(- 1, 'internal'));
```

Comment:
Can't see why this fix is fixing something. Can you point me ?

Reply:
To clone this "object"  (dol_clone()) generates an error when you have an "deposit" line in the origin invoice.
So I replaced '$cloneFrom" with the "$object" to avoid this problem.
It seems not to be needed to clone "$object" to merge contact lists.

Reply:
You mean dol_clone() return an error according to content of $object ?
This looks very strange. Which error do you have ?